Clean-up unnecessary comments and sout

louxiu opened the pull request:
1.Interface has comments already, so the comments in implements is
unnecessary
2. Replace sout with log

turt2live commented on src/main/java/net/glowstone/GlowOfflinePlayer.java:
```diff
@@ -181,7 +195,7 @@ public static OfflinePlayer deserialize(Map<String, Object> val) {
         } else {
             // use UUID
-            return Bukkit.getServer().getOfflinePlayer(UUID.fromString(val.get("UUID").toString()));
+            return Bukkit.getServer().getOfflinePlayer(UUID.fromString(val.get("uuid").toString()));
```
This is a breaking change that will not work.

turt2live commented on src/main/java/net/glowstone/GlowServer.java:
```diff
@@ -1132,7 +1138,17 @@ public OfflinePlayer getOfflinePlayer(String name) {
-        return new GlowOfflinePlayer(this, name);
+        OfflinePlayer result = getPlayerExact(name);
+        if (result == null) {
+            //probably blocking (1 per player per minute)
```
I don't think this metric is accurate...

m3rcuriel replied:
It was the metric I found <a href=http://wiki.vg/Mojang_API#UUID_-.3E_Profile_.2B_Skin.2FCape>here</a>.

turt2live replied:
The comment says `1 player per minute` when the link you provided says `same player once per minute`. Those are different metrics :P 

And don't we have an internal profile cache? 

Also indentation is off on the line below now.

m3rcuriel replied:
I don't see the indentation issue?

turt2live replied:
Line below

turt2live commented on src/main/java/net/glowstone/entity/meta/profile/PlayerProfile.java:
```diff
@@ -42,8 +42,10 @@ public PlayerProfile(String name, UUID uuid) {
     public PlayerProfile(String name, UUID uuid, List<PlayerProperty> properties) {
-        Validate.notNull(name, "name must not be null");
         Validate.notNull(uuid, "uuid must not be null");
+        if (name == null) {
```
This is illogically placed

dequis replied:
You mean it should be after the `Validate.notNull(properties, "properties must not be null");`?

turt2live replied:
Ideally, because it's functionality and not validation. I'd also like to hear why the check was removed :P

turt2live commented on src/main/java/net/glowstone/util/bans/UuidListFile.java:
```diff
@@ -69,11 +92,15 @@ private Entry(UUID uuid, String fallbackName) {
+            String name;
+            try {
+                name = ProfileCache.getProfile(uuid).getName();
+            } catch (NullPointerException e) {
```
I'm personally not letting this kind of exception to be caught. Handle null correctly because this may falsely catch.

m3rcuriel replied:
I'm sorry, what would be the optimal way to do that? (sorry)

turt2live replied:
An if statement

```
MyType obj = ...;
if (obj != null) {
    // ...
} else {
    // ...
}
```

m3rcuriel replied:
Oh okay. Right.
Thanks!

turt2live commented on src/main/java/net/glowstone/util/bans/UuidListFile.java:
```diff
@@ -32,6 +46,15 @@ public boolean containsUUID(UUID uuid) {
         return false;
     }
 
+    public boolean contains(PlayerProfile profile) {
```
This is not documented and therefore the implicit contract is broken. This should either be documented or better named.

turt2live commented:
@m3rcuriel You have several additional problems to the ones commented on above:

- File mode changes (revert these)
  - Includes a file mode change on an otherwise unmodified file
- Multiple minor changes that do not need to be present
  - Eg: Comments in otherwise untouched areas
  - Eg: Changing of existing validation messages
- Removal of documentation from several places

Please resolve the concerns as soon as possible.

turt2live commented:
@m3rcuriel Your pull request is no longer mergeable due to recent commits on the master branch. Additionally, there are several concerns outlined in my previous comment. Please address all concerns as soon as possible, thanks.

m3rcuriel commented:
Okay so it's now mergable and I think I fixed everything, however during testing I ended up surpassing the profile requests to the Mojang API so I'll finish testing in the morning and push.

turt2live commented:
@m3rcuriel Once again unmergeable :(

turt2live commented:
@m3rcuriel This pull request is considered unmergeable as of yesterday. Please correct the issue for further review (also please ping me once it is mergeable so that I can continue to review it). Thanks.

m3rcuriel commented:
Yeah. Sorry, I'll get it done Wednesday. Busy week.

turt2live commented:
@m3rcuriel Wednesday has past :(

turt2live commented:
@m3rcuriel This pull request is not mergeable. Please resolve the issue for further review.

turt2live commented:
@m3rcuriel This is a final warning: Your pull request is not mergable. Please resolve the issue for further review.

m3rcuriel commented:
@turt2live So sorry - jumped right back in. I lost my IntelliJ liscense and this PR slipped through the cracks. I found a slight problem with it, but I should have it done tomorrow 

*crosses fingers*

m3rcuriel commented:
@turt2live trying to keep you updated - there's a weird bug I haven't resolved but I'll work more tomorrow.

m3rcuriel commented:
@turt2live I'm not going to develop this any more until even a single PR is pulled into master. I hit a snag and couldn't fix it and I'm not willing to invent into a project not being actively updated. Feel free to close/remove this if you need to. I'll keep an eye though and come back if this project starts getting commits again. A good possibility for something to pull is my other PR, #521, which has been ready to pull for a good 8 months. I know you're waiting on SpaceManiac, and I understand that. Wish you luck.

deathcap commented:
@m3rcuriel I've [merged your PR](https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/commit/0538fe37681a1fc680b539f286f4975ed84b28d8) into [Glowstone++](https://github.com/GlowstonePlusPlus/GlowstonePlusPlus), can you elaborate on the weird bug? Glowstone++ might have it

louxiu committed:
Clean-up unnecessary comments and sout

1.Interface has comments already, so the comments in implements is
unnecessary
2. Replace sout with log

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/GlowstoneMC/Glowstone?pullRequest=533) <br/>All committers have signed the CLA.

aramperes commented on src/main/java/net/glowstone/net/handler/play/player/AdvancementTabHandler.java:
```diff
@@ -1,16 +1,19 @@
     public void handle(GlowSession session, AdvancementTabMessage message) {
         if (message.getAction() == AdvancementTabMessage.ACTION_CLOSE) {
-            System.out.println("Received AdvancementTabHandler (close)");
+            GlowServer.logger.log(Level.SEVERE, "Received AdvancementTabHandler (close)");
```
`Level.DEBUG`/`Level.INFO` perhaps?

louxiu replied:
done

louxiu commented:
@momothereal Oh, I didn't comprehend the `Level` correctly. I think `DEBUG`(Level.FINE?) is suitable.
